django/db/models/functions.py

```diff
@@ -83,6 +83,28 @@ def _paired(self, expressions):
         return ConcatPair(expressions[0], self._paired(expressions[1:]))
 
 
+class Greatest(Func):
+    """
+    Chooses the maximum non-null expression and returns it.
+    """
+    function = 'GREATEST'
+
+    def __init__(self, *expressions, **extra):
+        if len(expressions) < 1:
+            raise ValueError('Greatest must take at least one expression')
+        super(Greatest, self).__init__(*expressions, **extra)
+
+    def as_sqlite(self, compiler, connection):
+        self.coalesce()
+        return super(Greatest, self).as_sql(compiler, connection, function='MAX')
+
+    def coalesce(self):
+        # any null results in null for the whole expression, wrap with coalesce
+        expressions = [
+            Coalesce(expression, Value('')) for expression in self.get_source_expressions()]
+        self.set_source_expressions(expressions)
```

Comment:
One problem with the ifnull approach is that if all arguments are null, then the ifnull is returned, whereas the PostgreSQL implementation will output null.

I guess this is one place where we can't abstract away differences between backends in a sane way. What we should do is clearly document that the results might differ between backends, and also document how to write a custom Greatest implementation that guarantees similar results between backends.